### ISSUE TITLE
Support makePartial()

### ISSUE THREAD

**Issue by larsnystrom** [phpstan/phpstan-mockery#7]
It is possible to create a partial mock like this:

```
class SomeTest extends \PHPUnit\Framework\TestCase {
    public function createMock(): MyClass
    {
        return Mockery::mock(MyClass::class)->makePartial();
    }
}
```

However, phpstan will complain that `Method SomeTest::createMock() should return MyClass but returns Mockery\Mock.`.

**Comment by ondrejmirtes** [comment]
What does the method actually return? What can you call on the object?

**Comment by larsnystrom** [comment]
`makePartial()` creates a mock which by default defers calls to the original implementation. Here's the relevant part from the docs:

> A runtime partial test double, also known as a passive partial mock, is a kind of a default state of being for a mocked object.
>
> $mock = \Mockery::mock('MyClass')->makePartial();
> 
> With a generated partial, we assume that all methods will simply defer to the parent class (MyClass) original methods unless a method call matches a known expectation. If we have no matching expectation for a specific method call, that call is deferred to the class being mocked. Since the division between mocked and unmocked calls depends entirely on the expectations we define, there is no need to define which methods to mock in advance.

So the object that is returned has the same interface as the one returned by `Mockery::mock(MyClass::class)`. Calling `makePartial()` on the mock will just change the behavior of the mock, not its interface.

**Comment by ondrejmirtes** [comment]
Implemented: https://github.com/phpstan/phpstan-mockery/commit/8f3f0dc0bbc8c413224459a25f89a2cef5924871

And released as 0.11.3: https://github.com/phpstan/phpstan-mockery/releases/tag/0.11.3
